### DISCUSSION TITLE
Is controller advice still necessary if I'm using the CustomDataFetchingExceptionHandler

### DISCUSSION THREAD

**Discussion by ssona09** [Netflix/dgs-framework#1017] [Q&A] [answered]
Hi I have a custom handler as per the documentation here https://netflix.github.io/dgs/error-handling/

My custom exception is `NotFoundException`. In my data fetcher, I am checking that the list returned by the query is not empty, and if it is, I want to throw my custom exception `NotFoundException`.

Should I still have a controller advice to return a ResponseEntity for my custom exception or is that not necessary due to me having a custom exception with a specific error message?

```
@Component
public class CustomDataFetchingExceptionHandler implements DataFetcherExceptionHandler {
    private final DefaultDataFetcherExceptionHandler defaultHandler = new DefaultDataFetcherExceptionHandler();

    @Override
    public DataFetcherExceptionHandlerResult onException(DataFetcherExceptionHandlerParameters handlerParameters) {
        if(handlerParameters.getException() instanceof NotFoundException) {
            Map<String, Object> debugInfo = new HashMap<>();
            debugInfo.put("Value", "data not found %s", handlerparameters.getException().getMessage());

            GraphQLError graphqlError = TypedGraphQLError.INTERNAL.message("data not found")
                    .debugInfo(debugInfo)
                    .path(handlerParameters.getPath()).build();
            return DataFetcherExceptionHandlerResult.newResult()
                    .error(graphqlError)
                    .build();
        } else {
            return defaultHandler.onException(handlerParameters);
        }
    }
}
```

`
public class NotFoundException extends RunTimeException {
    private String msg;
    public NotFoundException(String msg) {
        super();
        this.msg = msg;
    }
}
`

**Comment by paulbakker** [comment, marked as answer]
You shouldn't need a controller advice. The only controller that's being used is the `DgsRestController` that actually handles the `/graphql` endpoint. Before that controller returns anything, your custom exception handler already did it's work, so there is nothing to do for an advice.
